katello/hooks/pre/30-upgrade.rb

```diff
@@ -34,9 +34,11 @@ def migrate_foreman
 end
 
 def postgresql_10_upgrade
+  start_postgresql
+  (name, owner, enconding, collate, ctype, privileges) = %x{runuser postgres -c 'psql -lt | grep -E "^\s+postgres"'}.chomp.split('|').map(&:strip)
   stop_services
   Kafo::Helpers.execute('yum -y install rh-postgresql10-postgresql-server')
-  Kafo::Helpers.execute('scl enable rh-postgresql10 -- postgresql-setup --upgrade')
+  Kafo::Helpers.execute("scl enable rh-postgresql10 \"PGSETUP_INITDB_OPTIONS='--lc-collate=\"#{collate}\" --lc-ctype=\"#{ctype}\"' postgresql-setup --upgrade\"")
```

Comment:
couple of questions/comments, all of which are probably from the "super picky" kind:
* before this used `--` between `scl enable …` and the actual command, now it doesn't, what does that mean?
* should we also pass `--locale=%{locale}` for consistency?
* I think the quoting is a bit off. the string will evaluate to `scl enable rh-postgresql10 "PGSETUP_INITDB_OPTIONS='--lc-collate="#{collate}" --lc-ctype="#{ctype}"' postgresql-setup --upgrade"` so the section that starts at `PGSETUP_INITDB_OPTIONS` actually ends after `--lc-collate`
I wonder if the following is easier to parse
```suggestion
  Kafo::Helpers.execute(%Q{scl enable rh-postgresql10 "PGSETUP_INITDB_OPTIONS='--lc-collate=#{collate} --lc-ctype=#{ctype}' postgresql-setup --upgrade"})
```
(I don't think we need to quote collate/ctype)

Reply:
THanks for the suggestion, I updated it to a version that drops even more quotes and works.

Reply:
answering myself: Note: if you use <command> consisting of multiple arguments, you either need to use quotes or the -- command separator. Everything that follows the separator will be considered a command or its argument.